src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs

```diff
@@ -251,76 +240,20 @@ string fieldOrPropertyName
 		/// diagnostics.
 		/// </summary>
 		private IEnumerable<Diagnostic> GatherDiagnostics(
-			SemanticModel model,
 			MutabilityInspector inspector,
+			ISymbol fieldOrProperty,
+			ITypeSymbol fieldOrPropertyType,
 			Location location,
 			bool isStatic,
-			bool isReadOnly,
-			ITypeSymbol fieldOrPropertyType,
-			string fieldOrPropertyName,
-			ExpressionSyntax initializationExpression,
-			bool isProperty,
-			bool isAutoImplementedProperty
+			string fieldOrPropertyName
 		) {
 			if ( !isStatic ) {
 				yield break;
 			}
 
-			if( isProperty && !isAutoImplementedProperty ) {
-				// non-auto-implemented properties don't hold their own state.
-				// We should never emit diagnostics for them.
-				yield break;
-			}
-
-			// Auto-implemented properties should be treated similar to fields:
-			// 1. They should not have a setter (i.e. isReadOnly)
-			// 2. Their type should be immutable
-			//    a. Unless an initializer ensures it isn't
-
-			if( !isReadOnly ) {
-				yield return CreateDiagnostic(
-					location,
-					fieldOrPropertyName,
-					fieldOrPropertyType.GetFullTypeNameWithGenericArguments(),
-					MutabilityInspectionResult.Mutable(
-						fieldOrPropertyName,
-						fieldOrPropertyType.GetFullTypeNameWithGenericArguments(),
-						MutabilityTarget.Member,
-						MutabilityCause.IsNotReadonly
-					)
-				);
-
-				// TODO: it'd probably be reasonable to not bail here. However
-				// we need to update the unsafestatics report because it counts
-				// the number of diagnostics that come out (it assumes at most one
-				// error per-field.)
-				yield break;
-			}
-
-			// Always prefer the type from the initializer if it exists because
-			// it may be more specific.
-			if( initializationExpression != null ) {
-				var initializerType = model.GetTypeInfo( initializationExpression ).Type;
-
-				// Fall back to the declaration type if we can't get a type for
-				// the initializer.
-				if( initializerType != null && !( initializerType is IErrorTypeSymbol ) ) {
-					fieldOrPropertyType = initializerType;
-				}
-			}
-
-			MutabilityInspectionResult result;
-
-			// When we know the concrete type as in "new T()" we don't have to
-			// be paranoid about mutable derived classes.
-			if ( initializationExpression is ObjectCreationExpressionSyntax ) {
-				result = inspector.InspectConcreteType( fieldOrPropertyType );
-			} else {
-				result = inspector.InspectType( fieldOrPropertyType );
-			}
+			var result = inspector.InspectMember( fieldOrProperty );
 
 			if ( result.IsMutable ) {
-				result = result.WithPrefixedMember( fieldOrPropertyName );

```

Comment:
Removed this to get the diagnostics to render correctly; makes sense because we pushed these steps into the MI so this is now being done for us.